akka-stream/src/main/scala/akka/stream/impl/fusing/ActorGraphInterpreter.scala

```diff
@@ -511,9 +530,23 @@ private[stream] class ActorGraphInterpreter(_initial: GraphInterpreterShell) ext
         false
     }
 
+  private def adjustEventLimit(shell: GraphInterpreterShell): Unit =
+    if (effectiveEventLimit < shell.shellEventLimit) effectiveEventLimit = shell.shellEventLimit
```

Comment:
Ah, sorry for reading the comparison the wrong way around. But still, I think that the configuration file setting should cap the total limit because this is the only weapon we have against unfairness: a huge fused graph would get very long stretches of CPU that cannot be interrupted. 200 events correspond to ca. 3µs on typical hardware today, high throughput applications may want to set it higher (and then also configure the Actor Dispatcher’s `throughput` setting which defaults to `5` messages only).

The HTTP server blueprint has roughly 50 ports, which at the default inputBuffer settings yields 800 events—maybe we set the default limit to 1000 and remove this adaptation logic?